include/uxr/agent/middleware/ced/CedMiddleware.hpp

```diff
@@ -28,123 +28,339 @@ class CedMiddleware : public Middleware
+     * @param xml               The XML that describes the new CedParticipant (unused).
+     * @return  true if both CedParticipant have the same Domain identifier, and false in other case.
+     */
     bool matched_participant_from_xml(
             uint16_t participant_id,
             int16_t domain_id,
             const std::string& xml) const override;
 
+    /**
+     * @brief Checks whether an existing CedTopic, identified by the topic_id, matches with a new CedTopic that would
+     *        result from the creation of a new one using the reference representation.
+     *        It is considered that the CedTopics match if both have the same topic name.
+     * @param topic_id  The existing CedTopic identifier.
+     * @param ref       The reference that represents the new CedTopic.
+     *                  It defines the CedGlobalTopic name.
+     * @return  true if both CedTopic have associated the same CedGlobalTopic, and false in other case.
+     */
     bool matched_topic_from_ref(
             uint16_t topic_id,
             const std::string& ref) const override;
 
+    /**
+     * @brief Checks whether an existing CedTopic, identified by the topic_id, matches with a new CedTopic that would
+     *        result from the creation of a new one using the XML representation.
+     *        It is considered that the CedTopics match if both have the same topic name.
+     * @param topic_id  The existing CedTopic identifier.
+     * @param xml       The XML that describes the new CedTopic.
+     *                  It defines the CedGlobalTopic name.
+     * @return  true if both CedTopic have associated the same CedGlobalTopic, and false in other case.
+     */
     bool matched_topic_from_xml(
             uint16_t topic_id,
             const std::string& xml) const override;
 
+    /**
+     * @brief Checks whether an existing CedDataWriter, identified by the datawriter_id, matches with the new
+     *        CedDataWriter that would result from the creation of a new one using the reference representation.
+     *        It is considered that the CedDataWriters match if both are associated to the same CedGlobalTopic.
+     * @param datawriter_id The existing CedDataWriter identifier.
+     * @param ref           The reference that represents the new CedDataWriter.
+     *                      It defines the name of the associated CedGlobalTopic.
+     * @return true if both CedDataWriter have associated the same CedGlobalTopic name
+     */
     bool matched_datawriter_from_ref(
             uint16_t datawriter_id,
             const std::string& ref) const override;
 
+    /**
+     * @brief Checks whether an existing CedDataWriter, identified by the datawriter_id, matches with the new
+     *        CedDataWriter that would result from the creation of a new one using the XML representation.
+     *        It is considered that the CedDataWriters match if both are associated to the same CedGlobalTopic.
```

Comment:
Are they just matching `CedGlobalTopic`? Do not `datawriter_id` participate in the matching?

Reply:
No, just the `CedGlobalTopic`.